app/components/button_component.rb

```diff
@@ -37,4 +37,18 @@ def css_class
   def icon_content
     render IconComponent.new(icon:) if icon
   end
+
+  def content
+    original_content = super
+    if original_content.present? && icon.present?
+      # Content templates may include leading whitespace, which interferes with the layout when an
+      # icon is present. This can be solved in CSS using Flexbox, but doing so for all buttons may
+      # have unintended consequences.
+      trimmed_content = original_content.sub(/^\s+/, '')
```

Comment:
```suggestion
      trimmed_content = original_content.lstrip
```
This works on multiple lines too?
```ruby
> "    \n\n\n   \naaaa".lstrip
=> "aaaa"
```

Reply:
Yeah, that's definitely better 👍 

For context, I went down a bit of a rabbit hole trying (unsuccessfully) to find some way to modify this method without having to deal with HTML safety, and this just happened to be where I'd landed.

Reply:
Updated in 43d9398d9.